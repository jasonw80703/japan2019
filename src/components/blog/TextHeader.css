.text-header {
  text-align: center;
  color: gray;
}

.text-header-divider {
  position: relative;
  margin-top: 0;
  border: none;
  height: 1.5px;
  width: 300px;
  background: gray;
  margin-bottom: 20px;
}