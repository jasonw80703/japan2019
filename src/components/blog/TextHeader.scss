@import '../../assets/styles/colours';

.text-header {
  text-align: left;
  color: $light-red;
  margin-top: 3%;
  margin-left: 10%;
}

.text-header-divider {
  position: relative;
  margin-top: 0;
  border: none;
  height: 2px;
  width: 80%;
  background: white;
  margin-bottom: 1.5%;
}
