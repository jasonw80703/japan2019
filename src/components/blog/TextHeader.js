import React from 'react';
import PropTypes from 'prop-types';
import './TextHeader.scss';

export default function TextHeader({ header }) {
  return (
    <>
      <h3 className="text-header">{header}</h3>
      <hr className='text-header-divider' />
    </>
  );
};

TextHeader.propTypes = {
  header: PropTypes.string.isRequired,
};
