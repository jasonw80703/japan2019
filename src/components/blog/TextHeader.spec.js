import React from 'react';
import renderer from 'react-test-renderer';
import TextHeader from './TextHeader';

describe('<TextHeader />', () => {
  const defaultProps = {
    header: 'Subsection XYZ',
  };

  it('renders as expected', () => {
    const tree = renderer
      .create(<TextHeader {...defaultProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});