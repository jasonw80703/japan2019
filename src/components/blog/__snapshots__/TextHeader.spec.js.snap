// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextHeader /> renders as expected 1`] = `
Array [
  <h3
    className="text-header"
  >
    Subsection XYZ
  </h3>,
  <hr
    className="text-header-divider"
  />,
]
`;
